src/Compilers/CSharp/Test/Semantic/Semantics/PrimaryConstructorTests.cs

```diff
@@ -17742,5 +17742,44 @@ void M()
             comp.VerifyDiagnostics(expected);
             comp.VerifyEmitDiagnostics(expected);
         }
+
+        [Fact]
+        [WorkItem("https://github.com/dotnet/roslyn/issues/67371")]
+        public void AssignToRefField_01()
+        {
+            var source = @"
+ref struct S1
+{
+    public ref int R1;
+
+    public S1(ref int x)
+    {
+        R1 = ref x;
+    }
+}
+
+ref struct S2
+{
+    public ref int R2;
+
+    public S2([System.Diagnostics.CodeAnalysis.UnscopedRef] ref int x)
+    {
+        R2 = ref x;
+    }
+}
+
+ref struct S3(ref int x)
+{
+    public ref int R3 = ref x;
+}
+
+ref struct S4([System.Diagnostics.CodeAnalysis.UnscopedRef] ref int x)
```

Comment:
Should we also have a test for a `scoped ref` primary constructor parameter which is ref-assigned to ref field in a constructor and in a field initializer?